src/sparsezoo/analysis/helpers.py

```diff
@@ -0,0 +1,258 @@
+    else:
+        return 0
+
+
+def is_sparse_layer(model: ModelProto, node: NodeProto) -> bool:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node whose sparsity is being checked
+    :return: true if node weights have any sparsity, False otherwise
+    """
+    return get_node_sparsity(model, node) > 0
+
+
+def is_four_block_sparse_layer(model: ModelProto, node: NodeProto) -> bool:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node whose four block sparsity is being checked
+    :return: true if node weights have any four block sparsity, False otherwise
+    """
+    return get_node_four_block_sparsity(model, node) > 0
+
+
+def is_quantized_layer(node: NodeProto) -> bool:
+    """
+    :param node: node whose quantized status is being checked
+    :return: true if node contains quantized weights, False otherwise
+    """
+    return node.op_type in ["QLinearConv", "ConvInteger", "MatMulInteger"]
+
+
+def get_node_four_block_sparsity(model: ModelProto, node: NodeProto) -> float:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node whose four block sparsity is being calculated
+    :return: four block sparsity of node
+    """
+
+    num_zero_blocks, num_total_blocks = get_node_four_block_sparsity_sizes(model, node)
+    if num_total_blocks == 0:
+        return 0.0
+
+    return num_zero_blocks / num_total_blocks
+
+
+def get_node_sparsity(model: ModelProto, node: NodeProto) -> float:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node whose sparsity is being calculated
+    :return: proportion of zeros in given node
+    """
+    num_zeros, param_size = get_node_sparsity_sizes(model, node)
+    if param_size == 0:
+        return 0.0
+
+    return num_zeros / param_size
+
+
+def is_parameterized_prunable_layer(model: ModelProto, node: NodeProto) -> bool:
+    """
+    :param model: model in which node's parameter exists
+    :param node: node being checked
+    :return: True if this node performs a operation that is parameterized and
+        prunable, False otherwise
+    """
+    return (
+        node.op_type
+        in [
+            "Conv",
+            "MatMul",
+            "Gemm",
+            "QLinearConv",
+            "QLinearMatMul",
+            "ConvInteger",
+            "MatMulInteger",
+            "Gather",
+        ]
+        and not get_layer_param(model, node) is None
+    )
+
+
+def get_layer_param(model: ModelProto, node: NodeProto) -> np.ndarray:
+    """
+    Finds parameter value of node (the node's weight)
+
+    :param model: model in which node's parameter exists
+    :param node: node to which parameter belongs to
+    :return: a numpy array of param value, None if not found
+    """
+
+    if node.op_type in ["Conv", "ConvInteger"]:
+        initializer_name = node.input[1]
+
+    elif node.op_type in ["QLinearConv"]:
+        initializer_name = node.input[3]
```

Comment:
Could use some clarification here. `get_zero_point()` seems to assume that the zero point for a layer is stored in `node.input[3]` and here the weights for a QLinearConv are stored in `node.input[3]`. Is there potential for an incorrect read?